enhancements/scheduling/scheduler-profiles.md

```diff
@@ -0,0 +1,208 @@
+---
+title: scheduling-profiles
+authors:
+  - "@damemi"
+reviewers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+approvers:
+  - "@maszulik"
+  - "@ingvagabund"
+  - "@deads2k"
+creation-date: 2020-11-11
+last-updated: 2020-11-11
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/624-scheduling-framework/kep.yaml"
+  - "/enhancements/kube-apiserver/audit-policy.md"
+replaces:
+superseded-by:
+---
+
+# Scheduling Profiles
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes a new field in the `schedulers.config.openshift.io/v1` API to allow
+configuration of enabled and disabled scheduler plugins via certain defined "profiles".
+
+This will allow users to configure pod scheduling beyond the default set of enabled plugins. It
+is effectively the equivalent to current scheduler Policy config, which has been deprecated.
+
+## Motivation
+
+As of Kubernetes 1.19, plugin profile configuration has replaced Policy files for the scheduler.
+We currently support setting scheduler predicates and priorities through the Policy API, and need
+to offer similar configuration with the new API in order to maintain a level of feature parity.
+
+However, the new plugin API is more expansive than the old API and provides more configuration possibilities
+than is necessary for most customers or reasonable to support. For those reasons we propose an
+abstracted representation of the most common plugin configurations for users to configure.
+
+### Goals
+
+1. define several common scheduler plugin configurations
+2. add a `profile` API field to `schedulers.config.openshift.io/v1` to specify which predetermined
+profile to use
+3. enable these profiles by kube-scheduler-operator
+
+### Non-Goals
+
+* configuration of individual scheduler plugins
+* platform support for custom schedulers or out-of-tree plugins
+
+## Proposal
+
+This enhancement proposes a new field in `schedulers.config.openshift.io/v1` named `profile`. This field
+specifies which of the following plugin configurations to enable on the default cluster scheduler.
+
+* `Default`: this will use the default scheduler profile as defined in the kube-scheduler's internal registry.
```

Comment:
Are we going to have equivalent profiles for the descheduler?

Reply:
Not sure what you mean by equivalent, because the descheduler doesn't have 1:1 mappings for strategies to scheduler plugins. There are some that match up, like affinity, and so it could be possible to tie some profiles together but that may be better as a future goal for now

Reply:
(For others' reference, linking the descheduler profiles doc I'm assuming you're referring to: https://docs.google.com/document/d/19fSk-9lqu1dDOS6qMEb0eiuG2qbRai9KbaxeS2ipdns/edit?ts=5fb41fff#)

Reply:
Equivalent mechanism - yes, not quite equivalent profiles.